eng/scripts/DownloadDevPackages.ps1

```diff
@@ -0,0 +1,50 @@
+param (
+    [Parameter(Mandatory=$True)]
+    [string] $WorkingDirectory,
+    [Parameter(Mandatory=$True)]
+    [string] $NupkgFilesDestination,
+    [string] $NugetSource="https://pkgs.dev.azure.com/azure-sdk/public/_packaging/azure-sdk-for-net/nuget/v3/index.json"
+)
+
+. (Join-Path $PSScriptRoot ".." common scripts common.ps1)
+
+$allPackages = Get-AllPkgProperties
+$trackTwoPackages = $allPackages.Where({ $_.IsNewSdk })
+
+Push-Location $WorkingDirectory
+$nugetPackagesPath = Join-Path $WorkingDirectory nugetPackages
+New-Item -Path $WorkingDirectory -Type "directory" -Name "nugetPackages"
+
+$packagesInFeed = Find-Package -Source $NugetSource -AllVersion -AllowPrereleaseVersions
```

Comment:
After a little local testing I think you can simply do the following:

```
$packageVersion = [AzureEngSemanticVersion]::ParseVersionString($package.Version)
$packageVersion.IsPreReleaseLabel = $false # clear prerelease so ToString strips it prerelease label.  
$packageVersionNoPrerelease $packageVersion.ToString()

For some reason I couldn't get this to work locally with the v3 feed but the v2 one worked correctly
#$NugetSource="https://pkgs.dev.azure.com/azure-sdk/public/_packaging/azure-sdk-for-net/nuget/v3/index.json"
$NugetSource="https://pkgs.dev.azure.com/azure-sdk/public/_packaging/azure-sdk-for-net/nuget/v2"

Install-Package -Name $package.Name -Source $NugetSource -AllowPrereleaseVersions -MinimumVersion "${packageVersionNoPrerelease}-alpha" -MaximumVersion "${packageVersionNoPrerelease }-alphb" -Destination "$nugetPackagesPath" -Force -SkipDependencies
```

This essentially will pull the latest alpha version for the package version that is checked into the repo. If there is no current new version yet it will fail.

Reply:
THis still requires using wildcards for the version which does not work.

Reply:
It is not using wildcards it is using a version range and it does work as I tested it locally. Can you help me understand why it doesn't?

Reply:
This method is still very unreliable. I does not get all the packages. It give an error `no match for package` for some packages wve though they are clearly present in the feed. There is also the issue of where it matches more that one version which cause it to fail with `Unable to install, multiple packages matched`

Reply:
```
# Install-Package requires a v2 nuget feed end-point
$NugetSource="https://pkgs.dev.azure.com/azure-sdk/public/_packaging/azure-sdk-for-net/nuget/v2"
...
  $packageVersion = [AzureEngSemanticVersion]::ParseVersionString($package.Version)
  $packageVersion.IsPreRelease = $false # clear prerelease so ToString strips it prerelease label.
  $packageVersionNoPrerelease = $packageVersion.ToString()

  # To workaround some older invalid packages (i.e. Iot->IoT renamed packages) start the alpha version range to start at 6/21
  Write-Verbose "Installing alpha version for $($package.Name) ${packageVersionNoPrerelease}"
  $installedPackage = Install-Package -Name $package.Name `
    -Source $NugetSource `
    -MinimumVersion "${packageVersionNoPrerelease}-alpha.202106" `
    -MaximumVersion "${packageVersionNoPrerelease}-alphb" `
    -Destination "$nugetPackagesPath" `
    -AllowPrereleaseVersions `
    -Force `
    -SkipDependencies `
    -ErrorAction Ignore

  if ($installedPackage) {
    Write-Host "Installed $($package.Name) $($installedPackages.Version)"
  }
  else {
    Write-Warning "Did not find any matching package alpha versions for $($package.Name) ${packageVersionNoPrerelease}"
  }
```

I tested that code locally and it works as expected without issues.

Reply:
Thanks. this works.